lldb/source/Commands/CommandCompletions.cpp

```diff
@@ -230,6 +232,9 @@ class SymbolCompleter : public Completer {
 
       // Now add the functions & symbols to the list - only add if unique:
       for (const SymbolContext &sc : sc_list) {
+        if (m_match_set.size() >= m_request.GetMaxNumberOfResultsToAdd())
```

Comment:
`>`? If I have 255 completions, and the max number of completions is 255, shouldn't that be valid?

Reply:
In this case at the last iteration `m_match_set.size() == 254`  before adding the last completion, so it will still be added...
if `m_match_set.size() > m_request.GetMaxNumberOfResultsToAdd()` it means that the number of completions added at the end will be bigger than the max number